### PR TITLE
add Sequential compatibility to rnn layers

### PR DESCRIPTION
**Pull request by szha**

### CODE REVIEW

File: python/mxnet/gluon/rnn/rnn_layer.py [hunk 1]
```diff
@@ -151,10 +151,52 @@ def begin_state(self, batch_size=0, func=ndarray.zeros, **kwargs):
             states.append(func(name='%sh0_%d'%(self.prefix, i), **info))
         return states
 
-    def forward(self, inputs, states):
+    def forward(self, inputs, states=None):
+        """Performs the RNN transformation for all time steps.
```
Comment (piiswrong) [inline python/mxnet/gluon/rnn/rnn_layer.py]: This is not visible. Put it in the class doc string of each layer

File: python/mxnet/gluon/rnn/rnn_layer.py [hunk 2]
```diff
@@ -165,8 +207,10 @@ def forward(self, inputs, states):
                 self.i2h_weight[i].shape = (self._gates*self._hidden_size, inputs.shape[2])
                 self.i2h_weight[i]._finish_deferred_init()
         if inputs.context.device_type == 'gpu':
-            return self._forward_gpu(inputs, states)
-        return self._forward_cpu(inputs, states)
+            out = self._forward_gpu(inputs, states) # output, state
+        out = self._forward_cpu(inputs, states)
```
Comment (piiswrong) [inline python/mxnet/gluon/rnn/rnn_layer.py]: ???

File: python/mxnet/gluon/rnn/rnn_layer.py (hunk 1 above)
Comment (szha) [inline python/mxnet/gluon/rnn/rnn_layer.py]: ok

File: python/mxnet/gluon/rnn/rnn_layer.py [hunk 3]
```diff
@@ -487,6 +502,7 @@ class GRU(_RNNLayer):
     >>> input = mx.nd.random_uniform(shape=(5, 3, 10))
     >>> h0 = mx.nd.random_uniform(shape=(3, 3, 100))
     >>> output, hn = layer(input, h0)
+    >>> output = layer(input) # zeros are used as begin state
```
Comment (piiswrong) [inline python/mxnet/gluon/rnn/rnn_layer.py]: ···
     >>> input = mx.nd.random_uniform(shape=(5, 3, 10))
     >>> # by default zeros are used as begin state
     >>> output = layer(input)
     >>> # manually specify begin state.
      >>> h0 = mx.nd.random_uniform(shape=(3, 3, 100))
      >>> output, hn = layer(input, h0)
···

File: python/mxnet/gluon/rnn/rnn_layer.py [hunk 4]
```diff
@@ -286,6 +293,8 @@ class RNN(_RNNLayer):
         The recurrent state's shape is `(num_layers, batch_size, num_hidden)`.
         If `bidirectional` is True, state shape will instead be
         `(2*num_layers, batch_size, num_hidden)`
+        If input recurrent state is None, zeros are used as default begin states,
```
Comment (piiswrong) [inline python/mxnet/gluon/rnn/rnn_layer.py]: Recurrent state shape -> Recurrent state.

and change the beginning sentence to The recurrent states is an NDArray with shape xxx.

File: python/mxnet/gluon/rnn/rnn_layer.py (hunk 3 above)
Comment (szha) [inline python/mxnet/gluon/rnn/rnn_layer.py]: Updated.

**Comment by szha** [comment]
The scala unit test error doesn't seem related to this PR.

**Pull request merged**